crates/cli/src/subcommands/call.rs

```diff
@@ -68,12 +66,37 @@ pub async fn exec(mut config: Config, args: &ArgMatches) -> Result<(), Error> {
     let builder = reqwest::Client::new().post(format!(
         "{}/database/call/{}/{}",
         config.get_host_url(server)?,
-        address,
+        address.clone(),
         reducer_name
     ));
     let auth_header = get_auth_header_only(&mut config, anon_identity, as_identity, server).await;
     let builder = add_auth_header_opt(builder, &auth_header);
+    let describe_reducer = util::describe_reducer(
+        &mut config,
+        address.clone(),
+        server.map(|x| x.to_string()),
+        reducer_name.clone(),
+        anon_identity,
+        as_identity.cloned(),
+    )
+    .await?;
+
+    // String quote any arguments that should be quoted
+    let arguments: Vec<String> = arguments
+        .filter(|args| args.len() == describe_reducer.schema.elements.len())
+        .unwrap_or_default()
```

Comment:
I think this is incorrect, but happens to fail in a way that doesn't affect UX. Probably we should be doing:
```rust
let arguments = arguments.unwrap_or_default();
anyhow::ensure!(arguments.len() == describe_reducer.schema.elements.len(), "Expected {} arguments but found {}", describe_reducer.schema.elements.len(), arguments.len());
```

Reply:
Note that this will swallow up the schema mismatch error message with a less-descriptive arg count error; you may want to write a better error message.

A case where (I think) your logic will go wrong: if I define a reducer that takes zero arguments, then try to call it with some arguments, your version will silently succeed with no args.

Reply:
I actually did this intentionally so that the reducer is still called if there is an argument count mismatch so that the user can see the error from SpacetimeDB, which includes the schema for the reducer. I think if you call a function in SpacetimeDB that is expecting no arguments, but you provide arguments, it will return an error:

```
PS C:\Users\Boppy\clockworklabs\my_new_project> spacetime call module say_hello argument
Error: Invalid arguments provided for reducer `say_hello` for database `module` resolving to address `5fc9c102c694a8bd7a96e3588bd0f760`.

The reducer has the following signature:
        say_hello()

Caused by:
    0: Response text: invalid arguments for reducer say_hello: .: trailing characters at line 1 column 2
    1: HTTP status client error (400 Bad Request) for url (http://127.0.0.1:3000/database/call/5fc9c102c694a8bd7a96e3588bd0f760/say_hello)
```

Reply:
Huh, I would've thought your `filter` call would return `None` if the length doesn't match the schema, then you'd unwrap to the empty vec, which would succeed the call. I'm a little confused what's going on here, but it looks like the behavior's correct, which is what matters.